File: CslaContrib/CSharp/CslaSrd/RuleBusinessBase.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Text;
using Csla;
using Csla.Validation;
using CslaSrd;
using CslaSrd.Validation;

namespace CslaSrd
{
    [Serializable()]
    public abstract class RuleBusinessBase<T> : BusinessBase<T> where T : RuleBusinessBase<T> , Csla.Core.IEditableBusinessObject
    {
        /// <summary>
        /// Provides a collection of all validation rules on the object that are broken, i.e., in an invalid state.
        /// </summary>
        public BrokenRulesCollection BrokenRules
        {
            get
            {
                return BrokenRulesCollection;
            }
        }

        /// <summary>
        /// Provides a collection of all validation rules on the object. 
        /// </summary>
        public PublicRuleInfoList Rules
        {
            get
            {
                return PublicRuleInfoList.GetList(base.ValidationRules.GetRuleDescriptions());
            }
        }
    }

}

File: branches/2010.11.001/CslaContrib/VB/CslaSrd/RuleBusinessBase.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections
Imports System.Collections.Generic
Imports System.Collections.ObjectModel
Imports System.Text
Imports Csla
Imports Csla.Validation
Imports CslaSrd
Imports CslaSrd.Validation

Namespace CslaSrd
	<Serializable()> _
	Public MustInherit Class RuleBusinessBase(Of T As {RuleBusinessBase(Of T), Csla.Core.IEditableBusinessObject})
		Inherits BusinessBase(Of T)
		''' <summary>
		''' Provides a collection of all validation rules on the object that are broken, i.e., in an invalid state.
		''' </summary>
		Public ReadOnly Property BrokenRules() As BrokenRulesCollection
			Get
				Return BrokenRulesCollection
			End Get
		End Property

		''' <summary>
		''' Provides a collection of all validation rules on the object. 
		''' </summary>
		Public ReadOnly Property Rules() As PublicRuleInfoList
			Get
				Return PublicRuleInfoList.GetList(MyBase.ValidationRules.GetRuleDescriptions())
			End Get
		End Property
	End Class

End Namespace
